cogs/fun.py

```diff
@@ -1,16 +1,272 @@
+                    await ctx.send(embed=why_em)
+                else:
+                    raise ServiceError(f"something went boom (http {why.status})")
+
+    @command(aliases=["rhash", "robothash", "rh", "rohash"])
+    async def robohash(self, ctx: Context, *, meme: str) -> None:
+        """text => robot image thing"""
+        try:
+            e = discord.Embed(colour=0x690E8)
+            meme = urllib.parse.quote_plus(meme)
+            e.set_image(url=f"https://robohash.org/{meme}.png")
+            await ctx.send(embed=e)
+        except Exception as e:
+            raise ServiceError(f"something broke: {e!s}")
+
+    async def get_answer(self, ans: str) -> str:
+        if ans == "yes":
+            return "Yes."
+        elif ans == "no":
+            return "NOPE"
+        elif ans == "maybe":
+            return "maaaaaaybe?"
+        else:
+            raise BadArgument("internal error: invalid answer lmaoo")
+
+    @command(aliases=["shouldi", "ask"])
+    async def yesno(self, ctx: Context, *, question: str) -> None:
+        """Why not make your decisions with a bot?"""
+        async with aiohttp.ClientSession() as session:
+            async with session.get("https://yesno.wtf/api", headers=self.user_agent) as meme:
+                if meme.status == 200:
+                    mj = await meme.json()
+                    ans = await self.get_answer(mj["answer"])
+                    em = discord.Embed(title=ans, description="And the answer to" f" {question} is this", colour=0x690E8,)
+                    em.set_image(url=mj["image"])
+                    await ctx.send(embed=em)
+                else:
+                    raise ServiceError(f"oof (http {meme.status})")
+
+    @command(aliases=["dadjoke", "awdad", "dadpls", "shitjoke", "badjoke"])
+    async def joke(self, ctx: Context) -> None:
+        """Dad joke simulator 3017, basically"""
+        async with aiohttp.ClientSession() as session:
+            async with session.get("https://icanhazdadjoke.com", headers=self.dadjoke) as jok:
+                if jok.status == 200:
+                    res = await jok.text()
+                    res = res.encode("utf-8").decode("utf-8")
+                    await ctx.send(f"`{res}`")
+                else:
+                    raise ServiceError(f"rip dad (http {jok.status})")
+
+    @command(aliases=["bofh", "techproblem"])
+    async def excuse(self, ctx: Context) -> None:
+        """
+        Bastard Operator from Hell excuses.
+
+        Source: http://pages.cs.wisc.edu/~ballard/bofh
+        """
+        async with aiohttp.ClientSession() as session:
+            async with session.get("http://pages.cs.wisc.edu" "/~ballard/bofh/excuses") as r:
+                data = await r.text()
+                lines = data.split("\n")
+                line = random.choice(lines)
+                await ctx.send(f"`{line}`")
+
+    @command()
+    async def neko(self, ctx: Context) -> None:
+        """Shows a neko"""
```

Comment:
Add `.` to the docstring
```suggestion
        """Shows a neko."""
```